main.js

```diff
@@ -0,0 +1,142 @@
+import axios from "axios";
+
+// 1. Data Fetchen
+// Create reference
+const list = document.getElementById("list-of-users");
+
+// Verwijzing naar error message
+const errorMessage = document.getElementById("error-message");
+
+// Asynchrone functie (normale syntax):
+async function fetchData(name) {
+    let URI = "https://restcountries.com/v2/all";
+
+    if (name) {
+        URI = "https://restcountries.com/v2/name/" + name;
+    } else {
+        URI = "https://restcountries.com/v2/all";
+    }
+
+    try {
+        errorMessage.textContent = "";
+        const response = await axios.get(URI);
+
+        // Leeg de lijst bij aanvang
+        list.replaceChildren();
+
+        // sorteren op populatie grootte van groot naar klein want dat vind ik leuker dan andersom zoals gevraagd
+        response.data.sort((a, b) => b.population - a.population);
+
+        // Map door de data heen
+        response.data.map((country) => {
```

Comment:
Hier zou je nog spreading kunnen toepassen, zodat je niet overal `country.` voor hoeft te zetten
```suggestion
        response.data.map(({ flags: { png }, name, currencies, population, capital }) => {
```